### DISCUSSION TITLE
[BUG] Mailbox.org: Can't playback audio after upload

### DISCUSSION THREAD

**Discussion by combatwombat95** [TechbeeAT/jtxBoard#953] [Q&A]
Hi,
when trying to replay audiofiles AFTER synchronizing with DAVx5 the app freezes and crashes or tells me, that it can't playback the audio.  Playback works fine inside the app before I press sync-botton or reopen the app.  After uploading I can't open the attached audio file with the upload device or another phone, that is connected to the same webDAV without this error.  The file is uploaded just fine though and I can download it over the web link. 

Tried this with all different audio formats and downloaded and installed over F-Droid and gplay-store. Same problem always. 

moto g62
Android 13
jtx 2.06 (or 2.05.06 from F-Droid)

**Comment by patrickunterwegs** [comment]
Hi @combatwombat95 ,
Would it be possible that you save such a broken entry as an .ics and attach it here?

**Comment by combatwombat95** [comment]
Hi,
thx for the quick reply.

I add it as zip file, since .ics doesn't get accepted.
[ics_file.zip](https://github.com/TechbeeAT/jtxBoard/files/12897268/ics_file.zip)

**Comment by patrickunterwegs** [comment]
Thanks @combatwombat95 !
That's interesting now, the .ics looks like that: 

```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:+//IDN bitfire.at//ical4android
BEGIN:VTODO
DTSTAMP:20231013T180617Z
UID:5978cd33-cd0c-4ad0-ad97-e352e0d8be3f
SEQUENCE:0
CREATED:20231013T180204Z
LAST-MODIFIED:20231013T180301Z
SUMMARY:Test
CLASS:PUBLIC
STATUS:COMPLETED
ATTACH;FMTTYPE=video/3gpp;X-LABEL="2023-10-13T20:01:52.516.3gp";FILENAME="2
 023-10-13T20:01:52.516.3gp":https://dav.mailbox.org/attachments/NC0zMy0yNC
 0xOA/2023-10-13T20:01:52.516.3gp
COMPLETED:20231013T180251Z
PERCENT-COMPLETE:100
PRIORITY:5
END:VTODO
END:VCALENDAR
```

So the actual attachment (the audio itself) was replaced by a link on mailbox.org. This is basically a good approach for real attachments, but there are the following problems now: 

- The link is only accessible when you're logged in. So even if jtx Board could from an URL (which is currently not supported), the download would fail as the url is protected with the username and password
- The URL itself contains a new line with another information, that seems to be completely wrong in it's place ("0xOA/2023-10-13T20:01:52.516.3gp"), this would be part of the URL, which it is obviously not... jtx Board recognises that it's a weblink (because of the https://), but then seems to crash, either because it's an invalid URL (with that extra line) or it is an actual bug (because an URL shouldn't be shown as a playback element).

I'm afraid I can only take care of the crash, for the rest, it would be up to mailbox.org...

**Comment by combatwombat95** [comment]
Well, the link leads to the correct download url and I can download the file with login information through my browser (android and windows). I just thought the DAVx5 app would be able to download it, and give the local path to jtx board. 
DAVx5 has the mailbox login info and also uploaded the files in the first place.  

I noticed mailbox.org stores the file under the path:    https://dav.mailbox.org/attachments/
that is different from the path for tasks:                       https://dav.mailbox.org/caldav/

So i guess it's because the tasks and the connected files are stored at different web paths and DAVx5 can't access "/attachments" to download them. Seems to be no way of adding that path manually, unfortunately,

**Comment by patrickunterwegs** [comment]
Hi @combatwombat95 ,

Basically your approach that you expect is right, it should be like that. I also discussed that with DAVx5, but currently it just doesn't work like that, DAVx5 just hands over the URL on the server without processing it.

It's something that will come in the future, but for now they have other things prioritized for DAVx5..
